skbio/io/format/blast6.py

```diff
@@ -0,0 +1,311 @@
+
+>>> df = io.read(fh, format="blast+6", into=pd.DataFrame,
+...           columns=['qseqid', 'pident', 'mismatch', 'length', 'gapopen',
+...                    'qend', 'bitscore', 'sstart'])
+>>> df # doctest: +NORMALIZE_WHITESPACE
+  qseqid  pident  mismatch  length  gapopen  qend  bitscore  sstart
+0   moaC  100.00         0     161        0   161       330       1
+1   moaC   99.38         1     161        0   161       329       1
+
+References
+----------
+.. [1] http://blastedbio.blogspot.com/2014/11/column-headers-in-blast-tabular-\
+and-csv.html
+.. [2] http://www.ncbi.nlm.nih.gov/books/NBK279675/
+"""
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import pandas as pd
+import numpy as np
+
+from skbio.io import create_format, BLAST6FormatError
+
+blast6 = create_format('blast+6')
+
+_possible_columns = {'qseqid': str, 'qgi': int, 'qacc': str, 'qaccver': str,
+                     'qlen': int, 'sseqid': str, 'sallseqid': str, 'sgi': int,
+                     'sallgi': int, 'sacc': str, 'saccver': str,
+                     'sallacc': str, 'slen': int, 'qstart': int, 'qend': int,
+                     'sstart': int, 'send': int, 'qseq': str, 'sseq': str,
+                     'evalue': float, 'bitscore': float, 'score': int,
+                     'length': int, 'pident': float, 'nident': int,
+                     'mismatch': int, 'positive': int, 'gapopen': int,
+                     'gaps': int, 'ppos': float, 'frames': str, 'qframe': int,
+                     'sframe': int, 'btop': int, 'staxids': str,
+                     'sscinames': str, 'scomnames': str, 'sblastnames': str,
+                     'sskingdoms': str, 'stitle': str, 'salltitles': str,
+                     'sstrand': str, 'qcovs': int, 'qcovhsp': int}
+_default_columns = ['qseqid', 'sseqid', 'pident', 'length', 'mismatch',
+                    'gapopen', 'qstart', 'qend', 'sstart', 'send',
+                    'evalue', 'bitscore']
+
+
+@blast6.reader(pd.DataFrame, monkey_patch=False)
+def _blast6_to_data_frame(fh, columns=None, default_columns=False):
+    if default_columns and columns is not None:
+        raise ValueError("`columns` and `default_columns` cannot both be"
+                         " provided.")
+    if not default_columns and columns is None:
+        raise ValueError("Either `columns` or `default_columns` must be"
+                         " provided.")
+    if default_columns:
+        columns = _default_columns
+
+    df = pd.read_csv(fh, na_values='N/A', sep='\t', header=None)
```

Comment:
add `keep_default_na=False` to avoid pandas recognizing other missing data representations (we only want "N/A" to be considered missing)